Failure decoding administrative region — iso_3166_1_alpha3 not found

#466 added `AdministrationRegion.CodingKeys` as a private enumeration. This enumeration is essential, because `AdministrationRegion.countryCodeAlpha3` doesn’t match the spelling of the `iso_3166_1_alpha3` key in the JSON-formatted response. Unfortunately, since `AdministrationRegion.CodingKeys` is private, decoding a route response that contains the administrative regions [fails](https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-ios/2125/workflows/527f078c-0b13-45bd-a814-e664be191abc/jobs/23300), as seen in mapbox/mapbox-navigation-ios#2730:

```
/Users/distiller/project/MapboxCoreNavigationTests/OfflineRoutingTests.swift:33: error: -[MapboxCoreNavigationTests.OfflineRoutingTests testOfflineDirections] : failed - Unexpected Failure: unknown(underlying: Swift.DecodingError.keyNotFound(CodingKeys(stringValue: "iso_3166_1_alpha3", intValue: nil), Swift.DecodingError.Context(codingPath: [CodingKeys(stringValue: "routes", intValue: nil), _JSONKey(stringValue: "Index 0", intValue: 0), CodingKeys(stringValue: "legs", intValue: nil), _JSONKey(stringValue: "Index 0", intValue: 0), CodingKeys(stringValue: "admins", intValue: nil), _JSONKey(stringValue: "Index 0", intValue: 0)], debugDescription: "No value associated with key CodingKeys(stringValue: \"iso_3166_1_alpha3\", intValue: nil) (\"iso_3166_1_alpha3\").", underlyingError: nil)))
<unknown>:0: error: -[MapboxCoreNavigationTests.OfflineRoutingTests testOfflineDirections] : Asynchronous wait failed: Exceeded timeout of 2 seconds, with unfulfilled expectations: "Calculate route offline".
/Users/distiller/project/MapboxCoreNavigationTests/OfflineRoutingTests.swift:46: error: -[MapboxCoreNavigationTests.OfflineRoutingTests testOfflineDirections] : failed - No route returned
Test Case '-[MapboxCoreNavigationTests.OfflineRoutingTests testOfflineDirections]' failed (2.134 seconds).
```

The fix is to either make the `AdministrationRegion.CodingKeys` enumeration public or implement the Codable methods explicitly instead of relying on the default implementation.

This regression was introduced in v1.1.0 and makes the library unusable with Valhalla-based endpoints and MapboxNavigationNative.

/cc @mapbox/navigation-ios @avi-c

--- a/Sources/MapboxDirections/AdministrationRegion.swift
+++ b/Sources/MapboxDirections/AdministrationRegion.swift
@@ -11,7 +11,7 @@ public struct AdministrationRegion: Codable, Equatable {
         case countryCode = "iso_3166_1"
     }
 
-    public var countryCodeAlpha3: String
+    public var countryCodeAlpha3: String?
     public var countryCode: String
 
     public init(countryCode: String, countryCodeAlpha3: String) {

